esmvalcore/preprocessor/_mask.py

```diff
@@ -270,18 +293,19 @@ def _mask_with_shp(cube, shapefilename):
     y_p_90 = np.where(y_p_0 == 90., y_p_0 - 1., y_p_0)
 
     # Build mask with vectorization
-    if cube.ndim == 2:
-        mask = shp_vect.contains(region, x_p_180, y_p_90)
-    elif cube.ndim == 3:
-        mask[:] = shp_vect.contains(region, x_p_180, y_p_90)
-    elif cube.ndim == 4:
-        mask[:, :] = shp_vect.contains(region, x_p_180, y_p_90)
-
-    # Then apply the mask
-    if isinstance(cube.data, np.ma.MaskedArray):
-        cube.data.mask |= mask
-    else:
-        cube.data = np.ma.masked_array(cube.data, mask)
+    for region in regions[0:10]:
+        if cube.ndim == 2:
+            mask = shp_vect.contains(region, x_p_180, y_p_90)
+        elif cube.ndim == 3:
+            mask[:] = shp_vect.contains(region, x_p_180, y_p_90)
+        elif cube.ndim == 4:
+            mask[:, :] = shp_vect.contains(region, x_p_180, y_p_90)
+
+        # Then apply the mask
+        if isinstance(cube.data, np.ma.MaskedArray):
+            cube.data.mask |= mask
+        else:
+            cube.data = np.ma.masked_array(cube.data, mask)
 
```

Comment:
so is this implementation working for your glaciated mask? I am a bit concerned by the looping over the 10 regions since for the land/sea all one needs is the 0th (first) region. Do these regions stack up to be the full, most accurate mask?

Reply:
Hi @valeriupredoi , thanks for looking into this. Instead of going into the details of the glaciated mask, I think the masking should be rewritten to make the code more modular. My point is that the selection of only the first region is not something general, and is different per NE mask. Therefore, I suggest to not proceed with this PR, but instead go for a new PR where we restructure the code, such that one can add new NE masks, without having to touch the existing the code.

Reply:
OKay the current implementation (that has been around for about 2 years) relies on the existence of *only* the NE land and NE ocean masks (in the masks aux dir) - I suggest, for ease of change and preserve the current functionality w/o the need to retest to keep this current implementation, only adding functionality for the glaciated mask (and the mask itself in the masks aux dir). What do you think? :beer: